src/tests/build.cmd

```diff
@@ -69,14 +70,10 @@ set __BuildNeedTargetArg=
 :Arg_Loop
 if "%1" == "" goto ArgsDone
 
-if /i "%1" == "/?"     goto Usage
-if /i "%1" == "-?"     goto Usage
-if /i "%1" == "/h"     goto Usage
-if /i "%1" == "-h"     goto Usage
-if /i "%1" == "/help"  goto Usage
-if /i "%1" == "-help"  goto Usage
-if /i "%1" == "--help" goto Usage
+@REM All arguments following this tag will be passed directly to msbuild (as unprocessed arguments)
+if /i "%1" == "--"                    (set processedArgs=!processedArgs! %1&shift&goto ArgsDone)
```

Comment:
Previously, handling of `--` seemed broken. It was just continuing the loop, which could then recognize and process arguments that were expected to be passed through. Chances are, such args were going unrecognized and by happenstance, they were still getting passed through. But this changes the behavior to match the Usage notes.